vertical-pod-autoscaler/enhancements/7862-cpu-startup-boost/README.md

```diff
@@ -0,0 +1,148 @@
+# AEP-7862: CPU Startup Boost
+
+<!-- toc -->
+- [AEP-7862: CPU Startup Boost](#aep-7862-cpu-startup-boost)
+  - [Summary](#summary)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+  - [Design Details](#design-details)
+    - [Workflow](#workflow)
+    - [API Changes](#api-changes)
+      - [Priority of `StartupBoost`](#priority-of-startupboost)
+      - [CPU startup boost and VPA autoscaling modes](#cpu-startup-boost-and-vpa-autoscaling-modes)
+    - [Validation](#validation)
+    - [Startup Boost and Readiness/Startup Probes](#startup-boost-and-readinessstartup-probes)
+    - [Failed in-place downsizes](#failed-in-place-downsizes)
+  - [Test Plan](#test-plan)
+  - [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Summary
+
+Long application start time is a known problem for more traditional workloads
+running in containerized applications, especially Java workloads. This delay can
+negatively impact the user experience and overall application performance. One
+potential solution is to provide additional CPU resources to pods during their
+startup phase, but this can lead to waste if the extra CPU resources are not
+set back to their original values after the pods are ready.
+
+This proposal allows VPA to boost the CPU request and limit of containers during
+the pod startup and to scale the CPU resources back down when the pod is `Ready`,
+leveraging the [in-place pod resize Kubernetes feature](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources).
+
+
+### Goals
+
+* Allow VPA to boost the CPU request and limit of a pod's containers when the pod during the pod startup (from creation time until it becomes `Ready`.)
+* Allow VPA to scale pods down to the original CPU resource values, [in-place](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources), as soon as their [`Ready`](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions) condition is true.
```

Comment:
I assume "original CPU resource values" should actually be whatever the value the VPA recommender has set as a recommendation? 
Or is the idea to scale back to what was defined in the Pod spec?

Reply:
The idea is to scale down to the `max` of those values. I wrote this in the Workflow section, but not here. Will update this section too.